### PR TITLE
[6.x] Adds missing options for PhpRedis

### PR DESCRIPTION
**Pull request by Wizofgoz**
This PR adds support for options on PhpRedis connections that were not allowed before. This includes specifying a serializer, scan option, and failover behavior for cluster connections.

The serializer option allows for using igbinary which can use significantly less storage space as documented [here](https://medium.com/@akalongman/phpredis-vs-predis-comparison-on-real-production-data-a819b48cbadb).

The scan option allows for customizing the behavior of scan commands as described [here](https://github.com/phpredis/phpredis#setoption).

The failover option allows for customizing the behavior of a cluster connection as described [here](https://github.com/phpredis/phpredis/blob/develop/cluster.markdown#automatic-slave-failover--distribution).

Replaces #31021

### CODE REVIEW

**Comment by Wizofgoz** [comment]
Strange, the tests are passing locally but not in the CI

**Review by darielopper — APPROVED** [review]

**Comment by omega123456** [comment]
The serializer option doesnt seem to be working. The $option variable you reference in vendor/laravel/framework/src/Illuminate/Redis/Connectors/PhpRedisConnector.php:101 is never set, so it will never get into the if block

File: src/Illuminate/Redis/Connectors/PhpRedisConnector.php [hunk 1]
```diff
@@ -97,6 +97,14 @@ protected function createClient(array $config)
             if (! empty($config['read_timeout'])) {
                 $client->setOption(Redis::OPT_READ_TIMEOUT, $config['read_timeout']);
             }
+
+            if (! empty($options['serializer'])) {
```
Comment (danijelk) [inline src/Illuminate/Redis/Connectors/PhpRedisConnector.php]: I just tested and confirm that $options isn't in this scope.

**Comment by danijelk** [comment]
Any status on this bug?

Would be a good idea to add OPT_COMPRESSION_LEVEL & OPT_COMPRESSION while we're on this missing options issue. I should be able to do a new PR tomorrow if no one else has the time @Wizofgoz  @driesvints

**Comment by driesvints** [comment]
@danijelk feel free to send in a pr. Make sure you add a thorough description it.

**Comment by Wizofgoz** [comment]
I briefly looked into it and it looks like I copy/pasted between `createRedisClusterInstance()` and `createClient()` not realizing that the variable names were different and my IDE didn't warn me. After correcting the variable name, a few tests were failing and I ran out of time looking into it so feel free to take a crack at it if you have time.

**Pull request merged**